### PR TITLE
Use id instead of ids for both multiple and single id. Add pagination to all entities except organisations

### PR DESCRIPTION
**Pull request by umangshah-js**

### CODE REVIEW

**Comment by ntallapa12** [comment]
#1 can you add a test case for facts and posts.
#2 add documentation on this new field

File: models/category.js [hunk 1]
```diff
@@ -12,19 +13,50 @@ class CategoryModel extends MongoBase {
...
+                response["data"] = result.results;
+                response["paging"] = {};
+                response["paging"]["next"] = result.next;
+                response["paging"]["hasNext"] = result.hasNext;
+                response["paging"]["previous"] = result.previous;
+                response["paging"]["hasPrevious"] = result.hasPrevious;
+                return response
             });
     }
+    getPagingObject(queryObj, sortBy, sortAsc, limit, next, previous) {
```
Comment (ntallapa12) [inline models/category.js]: if building paging object is same across all the models, then write it in a util and reuse it

File: models/claimant.js [hunk 2]
```diff
@@ -12,19 +13,50 @@ class ClaimantModel extends MongoBase {
...
-            .then((results) => {
+        const pagingObj = this.getPagingObject(query, sortBy, sortAsc, limit, next, previous);
+        const database = config.get('databaseConfig:databases:factcheck');
+        return Q(MongoPaging.find(this.collection(database),pagingObj))
+            .then((result) => {
                 this.logger.info('Retrieved the results');
-                return results;
+                result["data"] = result.results;
+                let response = {};
+                response["data"] = result.results;
```
Comment (ntallapa12) [inline models/claimant.js]: we use single quotes for all string representations

File: models/user.js [hunk 3]
```diff
@@ -12,23 +13,26 @@ class UserModel extends MongoBase {
...
-
+        const pagingObj = this.getPagingObject(query, sortBy, sortAsc, limit, next, previous);
         // get database from env config
         const database = config.get('databaseConfig:databases:core');
-
+        let pagingNew = {};
         // get all users
-        return Q(this.collection(database).find(query).toArray())
-            .then((users) => {
+        return Q(MongoPaging.find(this.collection(database),pagingObj))
```
Comment (ntallapa12) [inline models/user.js]: eslint should catch the space issue between the parameters here. try running eslint or "grunt build" command

File: models/category.js (hunk 1 above)
Comment (umangshah-js) [reply models/category.js]: done

File: models/claimant.js (hunk 2 above)
Comment (umangshah-js) [reply models/claimant.js]: done

File: models/user.js (hunk 3 above)
Comment (umangshah-js) [reply models/user.js]: grunt build didn't work but all errors in grunt test are fixed

**Pull request merged**